config/config_file.go

```diff
@@ -16,6 +16,12 @@ func (l KnownConfigLoader) Name() string {
 }
 
 func (l KnownConfigLoader) Configure(cfg *Config) error {
+	// Skip loading config file if some authentication is already explicitly
+	// configured directly in the config by a user.
+	// See: https://github.com/databricks/databricks-sdk-go/issues/304
+	if l.isAnyAuthConfigured(cfg) {
+		return nil
+	}
```

Comment:
This should only happen if `cfg.Profile == ""`. Otherwise it may not pick up what is needed from the config file.